Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -196,8 +210,14 @@ await container.UpsertItemAsync<TItem>(value, new PartitionKey(value.PartitionKe
 
             partitionKeyValue ??= id;
 
+            PatchItemRequestOptions patchItemRequestOptions = new();
+            if (etag != default && !string.IsNullOrWhiteSpace(etag))

```

Comment:
Maybe we should check this further up and if it is null or whitespace intentionally then give the user a `ArgumentNullException` might be hard to detect with `default`.